lib/ansible/modules/cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -494,52 +561,36 @@ def ensure_route_table_absent(connection, module):
 
     if lookup == 'tag':
         if tags is not None:
-            try:
-                route_table = get_route_table_by_tags(connection, vpc_id, tags)
-            except EC2ResponseError as e:
-                module.fail_json(msg="Error finding route table with lookup 'tag': {0}".format(e.message),
-                                 exception=traceback.format_exc())
-            except RuntimeError as e:
-                module.fail_json(msg=e.args[0], exception=traceback.format_exc())
+            route_table = get_route_table_by_tags(connection, module, vpc_id, tags)
         else:
             route_table = None
     elif lookup == 'id':
-        try:
-            route_table = get_route_table_by_id(connection, vpc_id, route_table_id)
-        except EC2ResponseError as e:
-            module.fail_json(msg="Error finding route table with lookup 'id': {0}".format(e.message),
-                             exception=traceback.format_exc())
+        route_table = get_route_table_by_id(connection, module, route_table_id)
 
     if route_table is None:
         return {'changed': False}
 
     # disassociate subnets before deleting route table
-    ensure_subnet_associations(connection, vpc_id, route_table, [], module.check_mode, purge_subnets)
-    try:
-        connection.delete_route_table(route_table.id, dry_run=module.check_mode)
-    except EC2ResponseError as e:
-        if e.error_code == 'DryRunOperation':
-            pass
-        else:
-            module.fail_json(msg="Error deleting route table: {0}".format(e.message),
-                             exception=traceback.format_exc())
+    if not module.check_mode:
+        ensure_subnet_associations(connection, module, route_table, [], False, purge_subnets)
+        try:
+            connection.delete_route_table(RouteTableId=route_table['RouteTableId'])
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e, msg="Error deleting route table")
 
     return {'changed': True}
 
 
-def get_route_table_info(route_table):
-
-    # Add any routes to array
-    routes = []
-    for route in route_table.routes:
-        routes.append(route.__dict__)
-
-    route_table_info = {'id': route_table.id,
-                        'routes': routes,
-                        'tags': route_table.tags,
-                        'vpc_id': route_table.vpc_id}
-
-    return route_table_info
+def get_route_table_info(connection, module, route_table):
+    result = get_route_table_by_id(connection, module, route_table['RouteTableId'])
+    try:
+        result['Tags'] = describe_tags_with_backoff(connection, route_table['RouteTableId'])
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Couldn't get tags for route table")
+    result = camel_dict_to_snake_dict(result, ignore_list=['Tags'])
```

Comment:
This is nice!

Reply:
Thanks,  #31400 (which this PR is currently based on) could use merging to allow this to happen for more modules.

Reply:
Oh, `ignore_list` is part of this PR. I thought I'd made it part of the other camel to snake PR. Never mind.